### PR TITLE
Edit objectives/bhbc 917

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to jira tickets

- {Include links to all of the applicable Jira tickets}

## This PR contains the following changes

- Edit Modal for component editing, ciurrent based on Project Objectives

## This PR contains the following types of changes

- [X] New feature (change which adds functionality)

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [x] I have performed a self-review of my own code

### Tests

- [x] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

Before edit:
Screen Shot 2021-03-23 at 11 01 10 AM

After edit (with alert displaying the data change)
Screen Shot 2021-03-23 at 11 01 31 AM

### CODE REVIEW

File: app/src/components/dialog/EditDialog.test.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+export interface ISampleFormikForm {
+  testField: string;
+}
+
+export const SampleFormikFormInitialValues: ISampleFormikForm = {
+  testField: ''
+};
+
+export const SampleFormikFormYupSchema = yup.object().shape({
+  testField: yup.string().max(3000, 'Cannot exceed 3000 characters').required('You must a test field for the project')
```
Comment (sdevalapurkar) [inline app/src/components/dialog/EditDialog.test.tsx:17]:
```suggestion
  testField: yup.string().max(3000, 'Cannot exceed 3000 characters').required('You must provide a test field for the project')
```

File: app/src/components/dialog/EditDialog.test.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        fullWidth
+        variant="outlined"
+        value={values.testField}
+        onChange={handleChange}
+      />
+    </form>
+  );
+};
+
+describe('ErrorDialog', () => {
```
Comment (sdevalapurkar) [inline app/src/components/dialog/EditDialog.test.tsx:43]:
```suggestion
describe('EditDialog', () => {
```

File: app/src/components/dialog/EditDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+   */
+  onCancel: () => void;
+  /**
+   * Callback fired if the 'Yes' button is clicked.
+   *
+   * @memberof IEditDialogProps
+   */
+  onSave: (values: IEditDialogComponentProps) => void;
+  //TODO
+  //Add interface for the values
```
Comment (sdevalapurkar) [inline app/src/components/dialog/EditDialog.tsx:59]: what is this TODO referring to? Don't we already have this?

File: app/src/components/dialog/EditDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+/**
+ * A dialog for displaying a title + message (typically a question), and giving the user the option to say
```
Comment (sdevalapurkar) [inline app/src/components/dialog/EditDialog.tsx:63]: this comment probably needs to be updated (seems like it is referring to the yes/no dialog)

File: app/src/constants/i18n.ts [hunk 5]
```diff
@@ -13,3 +13,11 @@ export const UploadProjectArtifactsI18N = {
   uploadErrorText:
     'An error has occurred while attempting to upload project artifacts, please try again. If the error persists, please contact your system administrator.'
 };
+
+export const EditObjectivesI18N = {
+  editTitle: 'Edit Project Objectives',
+  editText: 'Are you sure you want to save?',
+  createErrorTitle: 'Error Editing Project Objectives',
+  createErrorText:
```
Comment (sdevalapurkar) [inline app/src/constants/i18n.ts:21]: I don't see these being used anywhere - are they supposed to be?

File: app/src/components/dialog/EditDialog.test.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  it('matches the snapshot with no error message', () => {
```
Comment (sdevalapurkar) [inline app/src/components/dialog/EditDialog.test.tsx:44]: I dont see any error message being used or shown in the editdialog component - I think it needs to be added and the i18n values used in it and then we can update this test to show an error state and the commented test can become the no-error state test

**Comment by anissa-agahchen** [comment]
Too many merge conflicts ... will put changes onto new copy from dev

File: app/src/components/dialog/EditDialog.test.tsx (hunk 1 above)
Comment (anissa-agahchen) [reply app/src/components/dialog/EditDialog.test.tsx:17]: ok

File: app/src/components/dialog/EditDialog.test.tsx (hunk 2 above)
Comment (anissa-agahchen) [reply app/src/components/dialog/EditDialog.test.tsx:43]: ok

File: app/src/components/dialog/EditDialog.tsx (hunk 4 above)
Comment (anissa-agahchen) [reply app/src/components/dialog/EditDialog.tsx:63]: ok

File: app/src/constants/i18n.ts (hunk 5 above)
Comment (anissa-agahchen) [reply app/src/constants/i18n.ts:21]: yeah we need to add them, but we don't have error handling on this PR

**Pull request closed without merging**